include/deal.II/base/vector_slice.h

```diff
@@ -63,74 +68,14 @@ class VectorSlice
    */
   VectorSlice(VectorType &v, unsigned int start, unsigned int length);
 
+protected:
   /**
-   * Conversion operator to an ArrayView object that represents an array of
-   * non-const elements pointing to the same location as the current object.
-   */
-  operator ArrayView<typename VectorType::value_type *>();
-
-  /**
-   * Conversion operator to an ArrayView object that represents an array of
-   * const elements pointing to the same location as the current object.
-   */
-  operator ArrayView<const typename VectorType::value_type *>() const;
```

Comment:
I still cannot figure out these conversions: it should convert to `ArrayView<const typename VectorType::value_type>` (without the pointer). Was this always wrong or am I missing something obvious?

Reply:
I bet this operator was never instantiated. I suspect it would not have compiled. It might be fun to trace the history of when this was added and why :-)

Reply:
It showed up in #2010. You must be write regarding instantiation: the types don't make any sense.

Reply:
It was introduced in commit 089acc97800415197c494a885ffabc4ad3f5ccf7, @bangerth you are the author :-D

Reply:
> ~~write~~ right

@drwells You are clearly a native speaker. :rofl:

Reply:
I was really tired when I made the original mistake.....

Reply:
Ha, that's funny. I wonder if I had intended this to work via automatic casts, found that the compiler didn't want to do it (not because of a compiler or language problem, but because I was just stupid), and then used `make_array_view` by hand...